pkg/gofr/container/container_test.go

```diff
@@ -160,15 +160,16 @@ func TestContainer_Close(t *testing.T) {
 	controller := gomock.NewController(t)
 	defer controller.Finish()
 
-	mockDB := NewMockDB(controller)
+	mockDB, sqlMock, _ := gofrSql.NewSQLMocks(t)
+	e := expectedQuery{}
 	mockRedis := NewMockRedis(controller)
 	mockPubSub := &MockPubSub{}
 
-	mockDB.EXPECT().Close().Return(nil)
 	mockRedis.EXPECT().Close().Return(nil)
+	sqlMock.ExpectClose()
 
 	c := NewContainer(config.NewMockConfig(nil))
-	c.SQL = mockDB
+	c.SQL = &sqlMockDB{mockDB, &e, logging.NewLogger(logging.DEBUG)}
```

Comment:
Can we not directly write it like this? Is there any specific need of variable `e` ?

`c.SQL = &sqlMockDB{mockDB, &expectedQuery{}, logging.NewLogger(logging.DEBUG)}`

Reply:
I am initialising a struct and then I am passing the pointer to the same struct in mocksql as well as mocksqlDB. S I intialised it seperately. Though I will change e to a better variable name.